### ISSUE TITLE
BVH: Optimize preparation of geometry data.

### ISSUE THREAD

**Issue by Glidias** [Mugen87/yuka#19]
There are some minor side issues like the constructor parameters documentation ordering for BVH.js doesn't match the ordering of the function parameters.

Also, from fromMeshGeometry() method could consider additional approaches (in the event of already referencing non-indexed geometry for the parameter, to avoid cloning the Meshgeometry via the toPolygonSoup() call). I understand that cloning does provide stability in case the original mesh geometry may had been altered via some other apps, but I guess seperating out fromMeshGeometry() into two seperate functions (.. ie. another function that avoids calling toPolygonSoup() cloning/conversion process,  may be considered to allow bypassing additional cloning for certain application cases.

**Comment by Mugen87** [comment]
> There are some minor side issues like the constructor parameters documentation ordering for BVH.js doesn't match the ordering of the function parameters.

You're right, `primitivesPerNode` and `depth` are mixed. Fixed here 96e6e9f7735ddca25799078d39c08ac23c729f9a. I'll updating the website later.

`MeshGeometry.toTriangleSoup()` is currently only called in `BVH`. How about `BVH.fromMeshGeometry()` only performs the conversion if the geometry is indexed. In this way, the caller can pass in a non-indexed geometry and does not trigger further processing overhead. Something like:
```js
if ( geometry.indices !== null )  geometry = geometry.toTriangleSoup();
```

**Comment by Glidias** [comment]
That's what i did on my end as well. Even though there's a possible case (however rare) that people might want change the MeshGeometry data (whether intentionally or not), which will invalidate the BVH.

**Comment by Mugen87** [comment]
Solved via 328e3f12cc4442ea763b50456ec73f5906b32878.
